File: rb/features.rb
# DatagovCkan SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module DatagovCkanFeatures
  def self.make_feature(name)
    case name
    when "base"
      DatagovCkanBaseFeature.new
    when "test"
      DatagovCkanTestFeature.new
    else
      DatagovCkanBaseFeature.new
    end
  end
end

File: py/datagovckan_sdk/features.py
# DatagovCkan SDK feature factory

from datagovckan_sdk.feature.base_feature import DatagovCkanBaseFeature
from datagovckan_sdk.feature.test_feature import DatagovCkanTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DatagovCkanBaseFeature(),
        "test": lambda: DatagovCkanTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
